Makefile

```
@@ -135,9 +135,16 @@ doc:
 doc-spelling:
 	@make -C docs spelling SPHINXOPTS="-W -E"
 
-.PHONY: install
-install:
+.update-pip:
 	@pip install -U 'pip'
+
+.PHONY: compile-deps
+compile-deps: .update-pip
+	@pip install pip-tools
+	@pip-compile --allow-unsafe -q requirements/dev.in
+
+.PHONY: install
+install: .update-pip
 	@pip install -Ur requirements/dev.txt
```

The canonical use of the lockfiles is as follows:
```suggestion
	@pip install -r requirements/dev.in -c requirements/dev.txt
```